src/greatest.js

```diff
@@ -7,19 +7,21 @@ export default function greatest(values, compare = ascending) {
     let maxValue;
     for (const element of values) {
       const value = compare(element);
-      if (defined
+      if ((defined
           ? ascending(value, maxValue) > 0
-          : ascending(value, value) === 0) {
+          : ascending(value, value) === 0
+        ) && value !== null) {
```

Comment:
I think we should do `value != null`, and do this check before invoking ascending (or below, the passed-in comparator).

Reply:
✅